eyibra/Decentralized-Energy-Grid-Management-Platform
# Decentralized Energy Grid Management

## Overview
This project implements a decentralized energy grid management system using Clarity smart contracts and TypeScript tests. The system includes functionality for energy tokens, energy trading, grid management, and smart meter integration.

## Changes Made
- Implemented four main Clarity contracts: energy-token, energy-trading, grid-management, and smart-meter-integration
- Created corresponding TypeScript test files for each contract
- Simplified contract interactions to focus on core functionality
- Updated test files to use Vitest instead of Clarinet

## Files Created/Modified

### Clarity Contracts
1. `contracts/energy-token.clar`
2. `contracts/energy-trading.clar`
3. `contracts/grid-management.clar`
4. `contracts/smart-meter-integration.clar`

### TypeScript Tests
1. `tests/energy-token.test.ts`
2. `tests/energy-trading.test.ts`
3. `tests/grid-management.test.ts`
4. `tests/smart-meter-integration.test.ts`

## Implementation Details

### Energy Token Contract
- Implements basic token functionality (mint, transfer, burn)
- Includes balance and total supply tracking

### Energy Trading Contract
- Allows creation, cancellation, and fulfillment of energy sell orders
- Manages order details and status

### Grid Management Contract
- Tracks overall grid status (total supply and demand)
- Manages individual participant status
- Implements demand response functionality

### Smart Meter Integration Contract
- Handles registration and management of smart meters
- Allows updating meter readings and reporting energy production/consumption

## Testing Instructions
1. Ensure you have Node.js and npm installed
2. Install project dependencies:
